pkg/api/v1/helper/helpers.go

```diff
@@ -524,3 +524,28 @@ func StorageNodeAffinityToAlphaAnnotation(annotations map[string]string, affinit
 	annotations[v1.AlphaStorageNodeAffinityAnnotation] = string(json)
 	return nil
 }
+
+// NodeConditionEqual returns true if all effective types ("Status" is true) equals;
```

Comment:
Not just that, only the ones whose status is true is compared. This is fine because we only care about node condition updates from true to non-true (either false or unknown). I suggest renaming it to `NodeInSameContion`, change c1 & c2 to old & cur, and move it to be a private function under daemonset controller.